### ISSUE TITLE
Obtaining a certificate goes into an endless loop and fails.

### ISSUE THREAD

**Issue by constantineav** [rekby/lets-proxy2#118]
Getting some certificate goes into an endless loop and fails. I turned on the debug to figure it out. File in attachment. This does not happen with all certificates.
[lets-proxy.log](https://github.com/rekby/lets-proxy2/files/4301686/lets-proxy.log)

**Comment by constantineav** [comment]
Here is another good example. The 3rd level domain receives a certificate and the 4th level does not.
[lets-proxy2.log](https://github.com/rekby/lets-proxy2/files/4301721/lets-proxy2.log)

**Comment by rekby** [comment]
Hello.
Show section [CheckDomains] from your config please.

**Comment by constantineav** [comment]
Hello!
```
[CheckDomains]

# Allow domain if it resolver for one of public IPs of this server.
IPSelf = false

# Allow domain if it resolver for one of the ips.
IPWhiteList = ""

# Regexp in golang syntax of blacklisted domain for issue certificate.
#This list overrided by whitelist.
BlackList = ""

# Regexp in golang syntax of whitelist domains for issue certificate.
#Whitelist need for allow part of domains, which excluded by blacklist.
#
WhiteList = ""

# Comma separated dns server, used for resolve ip:port address of domains while check it.
# if empty - use system dns resolver (usually include hosts file, cache, etc)
# if set - use direct dns queries for servers, without self cache.
# if set more, than one dns server - send queries in parallel to all servers.
# error results from part of servers - ignore. Need minimum one answer.
# if different dns servers return different ip addresses - all of them use for check
# Example: "8.8.8.8:53,1.1.1.1:53,77.88.8.8:53,[2a02:6b8::feed:0ff]:53,[2001:4860:4860::8888]:53"
Resolver = ""

```

**Comment by rekby** [comment]
1. Program has mistake - as you say it infinite loop if can't authorize some of domains (in your situation www.subdomain). It fixed it in v0.23.10.
2. By default lets-proxy2 try issue two domains in a certificate: domain and www.domain. Usually if subdomain unavailable - it filtered by IP filter. You disable it and lets-proxy can't know about domain unexisted. Since v0.23.11 you can custom subdomains in [General] Subdomains. You can set it to [] (empty) for disable auto add subdomains. But you need enable filters, because everyone can connect to your server and say any domain in hello message. Then lets-proxy (without filter) will try to obtain certificate for the domain IssueTimeout (default 5 minutes). And your account can reach rate limit of failed authorisation and can't issue your normal certificates.
In v0.23.10 I add additional detections of self IP and I think it will good work with default filter settings in most cases.
If not - you can describe your situation and I can support it in auto mode or help with filter settings.

Please try v0.23.11 with IPSelf=true or with IP or your server in IPWhiteList.
